### DISCUSSION TITLE
Upgraded, low-cost camera for reduced image blur

### DISCUSSION THREAD

**Discussion by geezacoleman** [geezacoleman/OpenWeedLocator#17] [Ideas]
Current Raspberry Pi HQ camera is good for slow speed, but becomes blurry under speeds over 8-9 kmh. Any suggestions of relatively low-cost but high quality cameras that are compatible with the RPi would be very welcome.

**Comment by plains203** [comment]
https://www.raspberrypi.org/forums/viewtopic.php?t=278195 this link may be of interest. 

Something like this might be worth looking at. Not sure what resolution you need. 
https://www.arducam.com/product/arducam-full-hd-color-global-shutter-camera-for-raspberry-pi-2-3mp-ar0234-wide-angle-pivariety-camera-module-b0353/

**Comment by geezacoleman** [comment]
That's very handy - thanks for the suggestion. We're currently dropping the resolution down to 416 x 320 anyway, so the 3MP could work. We did have some issues with drivers for the arducam previously, but not something that's impossible to overcome if it fixes the issue.

**Comment by benboughton1** [comment]
When looking at the project I was surprised it worked at all with a rolling shutter on a moving platform!

I have played around with Arducam very early on but back then it required USB connectivity. Seeing now they link direct to the camera port and use hardware functions of pi is encouraging. 

I look forward to hearing if there will be a global shutter camera selected for this project.

**Comment by geezacoleman** [comment]
Hopefully will have that implemented in the future! I don't think rolling shutter is the largest issue here, it is one but the low light blur from longer shutter speeds I think is probably the most performance reducing factor. We do have some arducams that are global shutter and will attempt to get those added at some stage, but they weren't as plug and play as we had hoped.

**Comment by benboughton1** [comment]
What limits arducam plug and play?

**Comment by geezacoleman** [comment]
We had issues with working out drivers for the USB3.0 [board](https://www.arducam.com/product/usb3-0-camera-shield-2/). I'm sure it's very manageable to solve and something on the to do list, but we focused on the HQ camera for the initial version to keep things simple.

**Comment by williamtsalter** [comment]
We will soon be doing some testing of different camera solutions with an honours student.

So far we have identified the following cameras to test:
- **RPi HQ camera** (fully automatic exposure control)
- **RPi HQ camera** (with tweaked gain settings for better low light performance and reduced blur)
- **RPi V2 camera** (super low cost, plug and play alternative to HQ camera)
- **Arducam global shutter** (similar cost to HQ cam, recent updates from Arducam should make driver installation more accessible)
- **FLIR Blackfly S global shutter** (expensive, industry standard for machine vision applications)

**Reply by williamtsalter** [reply]
We will likely do the testing on Ubuntu 20 for RPi as there is extra support for the FLIR camera compared to Raspbian.

This will also make the transition to Jetson Nano (and potentially other Linux systems) simpler down the track. The current Jetson OS (Jetpack 4) is based on Ubuntu 18 but Nvidia are set to release Jetpack 5 in Q1 2022 (i.e. now!) based on Ubuntu 20, with full support for CUDA operations.

**Reply by benboughton1** [reply]
Is the RPi HQ still the pick of the cameras? I'm planning to assemble an OWL over the next 12 months and checking in on camera options.

**Reply by geezacoleman** [reply]
In terms of ease of use/install and current support the RPi HQ is still the best. But from a pure performance perspective it likely lags behind the RPi global shutter and the Arducam AR2034. Bit more discussion on the topic [here ](https://github.com/geezacoleman/OpenWeedLocator/discussions/92#discussioncomment-6849237)too.

I would really like to get the Global Shutter camera running on the RPi 4 or 5 because I reckon it will be much better, but it requires shifting to PiCamera2, which is entirely doable, but the repository/documentation lacks a bit of detail and I haven't had a chance to play around with it. That should change over the next 6 months once I get the PhD out the way.

If you're interested in relative performance you can see the results from the research above [here](https://arxiv.org/abs/2305.10311).

![image](https://github.com/geezacoleman/OpenWeedLocator/assets/51358498/299c7bc0-39da-4335-857b-9c75fe2627dd)

**Comment by benboughton1** [comment]
Great thanks! I have the Global Shutter (I've not used it) already but might go ahead and get the HQ as well.

Amazing how the arducam doesn't drop away nearly as much as the others at high speed. Global shutter more important than pixels it seems. Great research.

Sent from Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by geezacoleman** [reply]
Ah nice! Yeh the HQ will make it really easy to get started and hopefully once I get the software done you can switch to the global shutter.

Arducam was very impressive - we think it also involves pixel size on the sensor. Arducam is 9um, and RPi global shutter is approx. 11um. Increases the ability to 'accumulate more light for a given exposure time' - so improves quality. The answer [here ](https://photo.stackexchange.com/questions/122437/sensor-size-and-pixels-amount-impact-on-image-quality)is worth reading.

**Reply by ELIANVRANICICH** [reply]
Hello geezacoleman, I'm Elian Vranicich, from Argentina, I'm building a sprayer robot, using owl, I can't get arducam0234 to work, it's probably a driver issue, but I've run out of ideas, any info is appreciated, thank you very much for everything!

**Reply by geezacoleman** [reply]
Hey Elian! Awesome to hear you're building an OWL for  a sprayer robot. Keep us updated how you go and feel free to ask any questions if you run into issues. 

For this arducam problem, @williamtsalter was the person behind the Arducam set up last time, so he might have the resources he used. I do have the SD card handy, so can dive into that if needed. All I can remember is that it was quite difficult to get working reliably! We ran into some auto gamma correction issues when using it in the field. In saying that, the image quality and performance compared to the other cameras is astounding.

What sort of issues are you running into? This might help us work out how we can help.

Alternatively, the Raspberry Pi global shutter camera on the RPi5 could be an easier way to go if that's accessible to you? This just requires using picamera2.

**Comment by david-j-webster** [comment]
i have been trying to install the Rpi Global Shutter with no joy I keep getting to the start screen and get nothing. I have tried on both the Rpi 3+ and a jetson nano I can not get anything to work

**Reply by geezacoleman** [reply]
Thanks for giving this a go, and apologies it hasn't worked out.

I'm fairly sure this is a picamera vs. picamera2 issue. The Global Shutter doesn't work with picamera and OWL currently only works with picamera. Jetson Nano is another kettle of fish, but also something we hope to develop.

Support for picamera2 (therefore support for Global Shutter and Camera Module 3) and RPi5 is around the corner - April/May I expect to publish a working version of this. 

If you want to get the OWL working, I recommend either updating your code to Picamera2 or using an older HQ camera with Picamera.
